vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_unittest.cc

```diff
@@ -65,31 +65,259 @@ class PromotionTest : public testing::Test {
         std::make_unique<bat_ledger::MockLedgerImpl>(mock_ledger_client_.get());
     promotion_ = std::make_unique<Promotion>(mock_ledger_impl_.get());
   }
+};
+
+TEST_F(PromotionTest, TestInitialize) {
+  // Arrange
+  ON_CALL(*mock_ledger_client_, GetBooleanState(
+    ledger::kStatePromotionCorruptedMigrated))
+     .WillByDefault(testing::Return(false));
+
+  EXPECT_CALL(*mock_ledger_impl_, GetAllPromotions(_)).Times(2);
```

Comment:
@NejcZdovc `promotion->Initialize()` invokes the `GetAllPromotions` callback twice in this scenario, once to perform the `CheckForCorrupted` and then it falls through to invoke `Retry`... is this the expected behaviour? As it means `CheckForCorrupted` is invoking an async `LoadURL`, and then the code moves on to processing local data in `Retry`

Reply:
we could make it sync just for double safety

Reply:
@NejcZdovc do you want me to make that change as part of this PR? (I'd say it's a little out of scope tbh and is better as a new issue IMHO).

Reply:
- [ ] Ticket for GetAllPromotions rethink

Reply:
new issue is fine